lib/routes/shared/initial_walkthrough.dart

```diff
@@ -89,7 +103,10 @@ class InitialWalkthroughPageState extends State<InitialWalkthroughPage>
         widget._backupBloc.restoreFinishedStream.listen((restored) {
       popToWalkthrough();
       if (restored) {
-        _proceedToRegister();
+        widget._registrationBloc.userActionsSink.add(_updateSecurityModelAction);
+        _updateSecurityModelAction.future.then((_) {
+          _proceedToRegister();
+        });
       }
     }, onError: (error) {
       popToWalkthrough(
```

Comment:
When en error happens, for example when the phrase wasn't correct, the user would have to choose the restore snapshot again and re-enter the phrase again, which is tedious and bad experience.
Better NOT to pop the phrase/pin screens when restoring and push the loader route on top of them.
On error just pop the loader route and let the user change the pin/phrase or go back and start all over again.